pkgs/by-name/de/devmode/package.nix

```diff
@@ -1,123 +1,122 @@
 {
-  lib,
-  findutils,
-  nodejs_latest,
+  live-server,
   parallel,
-  rsync,
   watchexec,
-  writeShellScriptBin,
+  writeShellApplication,
   # arguments to `nix-build`, e.g. `"foo.nix -A bar"`
   buildArgs ? "",
   # what path to open a browser at
   open ? "/index.html",
 }:
 let
-  inherit (nodejs_latest.pkgs) live-server;
+  error-page = writeShellApplication {
+    name = "error-page";
+    text = ''
+      cat << EOF
+      <!DOCTYPE html>
+      <html>
+      <head>
+        <style>
+          @media (prefers-color-scheme: dark) {
+            :root { filter: invert(100%); }
+          }
+        </style>
+      </head>
+      <body><pre>$1</pre></body>
+      </html>
+      EOF
+    '';
+  };
 
-  error-page = writeShellScriptBin "error-page" ''
-    cat << EOF
-    <!DOCTYPE html>
-    <html>
-    <head>
-      <style>
-        @media (prefers-color-scheme: dark) {
-          :root { filter: invert(100%); }
-        }
-      </style>
-    </head>
-    <body><pre>$1</pre></body>
-    </html>
-    EOF
-  '';
+  build-and-link = writeShellApplication {
+    name = "build-and-link";
+    runtimeInputs = [ error-page ];
+    text = ''
+      set -euxo pipefail
 
-  # The following would have been simpler:
-  # 1. serve from `$serve`
-  # 2. pass each build a `--out-link $serve/result`
-  # But that way live-server does not seem to detect changes and therefore no
-  # auto-reloads occur.
-  # Instead, we copy the contents of each build to the `$serve` directory.
-  # Using rsync here, instead of `cp`, to get as close to an atomic
-  # directory copy operation as possible. `--delay-updates` should
-  # also go towards that.
-  build-and-copy = writeShellScriptBin "build-and-copy" ''
-    set -euxo pipefail
+      set +e
+      stderr=$(2>&1 nix-build --no-out-link ${buildArgs})
+      exit_status=$?
+      set -e
 
-    set +e
-    stderr=$(2>&1 nix-build --out-link $out_link ${buildArgs})
-    exit_status=$?
-    set -e
+      rm -rf "''${serve:?}"
 
-    if [ $exit_status -eq 0 ];
-    then
-      # setting permissions to be able to clean up
-      ${lib.getExe rsync} \
-        --recursive \
-        --chmod=u=rwX \
-        --delete-before \
-        --delay-updates \
-        --links \
-        $out_link/ \
-        $serve/
-    else
-      set +x
-      ${lib.getExe error-page} "$stderr" > $error_page_absolute
-      set -x
+      if [ $exit_status -eq 0 ];
+      then
+        out_link="$(nix-build --no-out-link ${buildArgs})"
```

Comment:
This would evaluate twice here, which may waste l quite a lot of time. Why not run it once with `--out-link ${serve:?}` and pipe stderr to the error file directly?

Reply:
Whoops, must've done something wrong when rewriting the commits (I dropped the commit which removes the error fallback function, causing a all kinds of conflicts...)